include/dev/hallSensor.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <EVT/io/GPIO.hpp>
+#include <EVT/DEV/RTCTimer.hpp>
+
+namespace IO = EVT::core::IO;
+namespace DEV = EVT::core::DEV;
+
+
+namespace hallSensor {
+
+class HallSensor {
+
+public:
+    enum WheelSpeedState {
+        STOPPED = 0,    // First pulse or wheel is dead
+        MAINTAIN = 1,   // Wheel is spinning at a constant speed or speeding up
+    };
+
+    // Constructor (take a GPIO instance and the radius of the wheel)
+    HallSensor(IO::GPIO& gpio, uint32_t wheelRadius);
+
+    uint32_t update(); // Update the wheel speed
+
+    // Get the current speed of the wheel
+    uint32_t getSpeed(uint32_t timeDiff);
+
+
+private:
+    IO::GPIO& gpio;            // GPIO instance for the sensor
```

Comment:
This is how we do one-line Doxygen commments. Also, a more descriptive name would be preferable
```suggestion
    /** GPIO instance for the sensor */
    IO::GPIO& sensorGPIO;  
```